homeassistant/components/microbees/cover.py

```diff
@@ -0,0 +1,115 @@
+"""Cover integration microBees."""
+from threading import Timer
+from typing import Any
+
+from microBeesPy import Actuator
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MicroBeesUpdateCoordinator
+from .entity import MicroBeesEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Config entry."""
+    coordinator: MicroBeesUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ].coordinator
+
+    async_add_entities(
+        MBCover(
+            coordinator,
+            bee_id,
+            next(filter(lambda x: x.deviceID == 551, bee.actuators)).id,
+            next(filter(lambda x: x.deviceID == 552, bee.actuators)).id,
+        )
+        for bee_id, bee in coordinator.data.bees.items()
+        if bee.productID == 47
+    )
+
+
+class MBCover(MicroBeesEntity, CoverEntity):
+    """Representation of a microBees cover."""
+
+    def __init__(self, coordinator, bee_id, actuator_up_id, actuator_down_id) -> None:
+        """Initialize the microBees cover."""
+        super().__init__(coordinator, bee_id)
+        self.actuator_up_id = actuator_up_id
+        self.actuator_down_id = actuator_down_id
+        self.attr_supported_features = {
+            CoverEntityFeature.OPEN,
+            CoverEntityFeature.STOP,
+            CoverEntityFeature.CLOSE,
+        }
+        self._attr_is_closed = None
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+
+    @property
+    def name(self) -> str:
+        """Name of the cover."""
+        return self.bee.name
+
+    @property
+    def actuator_up(self) -> Actuator:
+        """Return the rolling up actuator."""
+        return self.coordinator.data.actuators[self.actuator_up_id]
+
+    @property
+    def actuator_down(self) -> Actuator:
+        """Return the rolling down actuator."""
+        return self.coordinator.data.actuators[self.actuator_down_id]
+
+    def reset_open_close(self):
+        """Reset the opening and closing state."""
+        self._attr_is_opening = False
+        self._attr_is_closing = None
+        self.async_write_ha_state()
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        sendCommand = await self.coordinator.microbees.sendCommand(
+            self.actuator_up_id,
+            self.actuator_up.configuration.actuator_timing * 1000,
+        )
+        if sendCommand:
+            self._attr_is_opening = True
+            Timer(
+                self.actuator_up.configuration.actuator_timing,
+                self.reset_open_close,
+            ).start()
```

Comment:
Invert the if and raise if the sendCommand failed. Then we can outdent.

Why you use `Timer` here? You could make use of `async_call_later`, which is a helper function.